gwdetchar/scattering/__main__.py

```diff
@@ -0,0 +1,173 @@
+                             'default: %(default)s')
+    parser.add_argument('-o', '--output-dir', type=os.path.abspath,
+                        default=os.curdir,
+                        help='Output directory for analysis, '
+                             'default: %(default)s')
+    parser.add_argument('-c', '--colormap', default='viridis',
+                    help='name of colormap to use, default: %(default)s')
+
+    # parse arguments
+    args = parser.parse_args(args)
+    ifo = args.ifo
+    gps = float(args.gpstime)
+    gpsstart = gps - 0.5 * args.duration
+    gpsend = gps + 0.5 * args.duration
+    primary = ':'.join([ifo, args.primary_channel])
+    thresh = args.frequency_threshold
+    multipliers = [int(x) for x in args.multipliers.split(',')]
+    harmonic = args.multiplier_for_threshold
+    if '{IFO}' in args.primary_frametype:
+        args.primary_frametype = args.primary_frametype.format(IFO=ifo)
+    if '{IFO}' in args.aux_frametype:
+        args.aux_frametype = args.aux_frametype.format(IFO=ifo)
+    logger.info('{0} Scattering: {1}'.format(ifo, gps))
+
+    # set up spectrogram
+    logger.debug('Setting up a Q-scan spectrogram of {}'.format(primary))
+    hoft = get_data(primary, start=gps-34, end=gps+34,
+                    frametype=args.primary_frametype,
+                    verbose='Reading primary channel:'.rjust(30))
+    hoft = highpass(hoft, f_low=thresh).resample(256)
+    qspecgram = hoft.q_transform(qrange=(4, 150), frange=(0, 60), gps=gps,
+                                 fres=0.1, outseg=(gpsstart, gpsend), **ASD_KW)
+    qspecgram.name = primary
+
+    # process channels
+    channels = [':'.join([ifo, c]) for c in MOTION_CHANNELS]
+    data = get_data(                                                      
+        channels, start=gpsstart, end=gpsend, frametype=args.aux_frametype,  
+        verbose='Reading auxiliary sensors:'.rjust(30))
+    count = 0  # running count of plots written
+    for channel in channels:
+        logger.info(' -- Processing {} -- '.format(channel))
+        try:
+            motion = data[channel].detrend().resample(128)
+        except KeyError:
+            logger.warning('Skipping {}'.format(channel))
+            pass
+        # project scattering frequencies
+        fringe = get_fringe_frequency(motion, multiplier=1)
+        ind = fringe.argmax()
+        fmax = fringe.value[ind]
+        tmax = fringe.times.value[ind]
+        logger.debug('Maximum scatter frequency {0:.2f} Hz at GPS second '
+                     '{1:.2f}'.format(fmax, tmax))
+        if harmonic * fmax < thresh:
+            logger.warning('No significant evidence of scattering '
+                           'found in {}'.format(channel))
+            continue
+        # plot spectrogram and fringe frequency
+        output = os.path.join(
+            args.output_dir,
+            '%s-%s-%s-{}.png' % (
+                channel.replace(':', '-'), gps, args.duration)
```

Comment:
I think this needs to be a little more specific:
```suggestion
                channel.replace('-', '_').replace(':', '-', 1), gps, args.duration)
```